gluon-mesh-vpn-wireguard-vxlan/README.md

```diff
@@ -0,0 +1,38 @@
+# gluon-mesh-vpn-wireguard-vxlan
+
+You can use this package for connecting with wireguard to the Freifunk Munich network.
+
+You should use something like the following in the site.conf:
+
+        
+```
+ mesh_vpn = {
+        mtu = 1400,
```

Comment:
I think a short desription on which MTU to use is helpful.

Reply:
I think the minimum MTU calculation for this package goes like this:

| MTU (Bytes) | Header (Bytes) | Description | Reference |
|-|-|-|-|
| 1280 |  | Minimum for IPv6 | RFC2460 |
| 1312 | +32 | batman-adv | https://www.open-mesh.org/projects/batman-adv/wiki/Quick-start-guide |
| 1320 | +8 | VXLAN | RFC7348 |
| 1328 | +8 | UDP | https://en.wikipedia.org/wiki/User_Datagram_Protocol#UDP_datagram_structure |
| 1368 | +40 | IPv6 | RFC2460 | 
| 1448 | +80 | Wireguard | https://lists.zx2c4.com/pipermail/wireguard/2017-December/002201.html |

Reply:
Or to summarize, +13.125% (1280->1448, +168) vs. +9.1875% for fastd (1280->1398, +118), right?

So in the worst case, a system that had sufficient CPU for fastd previously might see a 3.45% performance decrease with Wireguard+vxlan (1398/1448) due to the increased header size?

Reply:
Which in my opinion seems absolutely ok, and might make it interesting to have Wireguard+vxlan as a default setting over fastd, if this works reliably.

Even though I know at least one setup at Freifunk Lübeck with a TP-Link 1043nd behind a 6 MBit/s ADSL line, where fastd is not the bottleneck but the DSL instead. So ideally would be nice if this were configurable in the config-mode.

Reply:
We only see increase of speeds with wireguard. It seems that the increased header has less impact on the CPU than the tunneling through userspace and back.

Reply:
Why does this table count _up_ from 1280? Shouldn't you be counting down from 1500?

MTU: 1500-80-40-8-8-32=1332

Or am I missing something fundamental about how you guys think about this?

Reply:
We started counting up because most of the devices are behind pppoe links or stuff with even lower MTU. So we assume the worst and count up from there, to get an idea of  how much headroom is left.

Reply:
@awlx What was the lowest throughput internet connection you tested this with so far?

For the 841 v9 @27 MBit/s, there the internet connection wasn't the bottleneck before, right?

Reply:
PS: Hope I don't sound like nit-picking on this. I'm just wondering if there might be some rare cases where fastd could be faster. Like when the DSL uplink was the bottleneck before and not CPU. And if for such rare cases it could be useful if the user were able to choose between fastd and wireguard in the web-ui.

Reply:
Problem is the 841 and other low in RAM/Flash targets can't even handle wireguard with all the tools. And so they cannot handle both vpn providers.

We didn't test yet a connection where the CPU was not the limiting factor. And I am not aware of anyone doing so at the moment.

If you could provide some results that would be awesome.